tests/test_trusted_metadata_set.py

```diff
@@ -110,13 +128,269 @@ def test_update_with_invalid_json(self):
+            true_hashes, b'{""sig": }'
+        )
+        self.trusted_set.timestamp.signed.meta["snapshot.json"].hashes = modified_hashes
+        with self.assertRaises(exceptions.RepositoryError):
+            self.trusted_set.update_snapshot(b'{""sig": }')
+        self.trusted_set.timestamp.signed.meta["snapshot.json"].hashes = true_hashes
+
+    def test_update_snapshot_cannot_verify_snapshot_with_threshold(self):
+        self._setup_update_snapshot_or_timestamp_test()
+        # root data with threshold which cannot be verified for new_snapshot
+        self.trusted_set.root.signed.roles["snapshot"].threshold = 2
+        with self.assertRaises(exceptions.UnsignedMetadataError):
+            self.trusted_set.update_snapshot(self.metadata["snapshot"])
+        self.trusted_set.root.signed.roles["snapshot"].threshold = 1
+
+    def test_update_snapshot_version_different_timestamp_snapshot_version(self):
+        self._setup_update_snapshot_or_timestamp_test()
+        # new_snapshot.version != trusted timestamp.meta["snapshot"].version
+        self.trusted_set.timestamp.signed.meta["snapshot.json"].version = 2
+        with self.assertRaises(exceptions.BadVersionNumberError):
+            self.trusted_set.update_snapshot(self.metadata["snapshot"])
+        self.trusted_set.timestamp.signed.meta["snapshot.json"].version = 1
+
+
+    def _setup_update_snapshot_after_successful_update_test(self):
+        self._setup_update_snapshot_or_timestamp_test()
+        self.trusted_set.update_snapshot(self.metadata["snapshot"])
+
+    def test_update_snapshot_after_successful_update_new_snapshot_no_meta(self):
+        self._setup_update_snapshot_after_successful_update_test()
+        # Test removing a meta_file in new_snapshot compared to the old snapshot
+        snapshot_obj = Metadata.from_bytes(self.metadata["snapshot"])
+        snapshot_obj.signed.meta = {}
+        # prepare timestamp.meta["snapshot"].hashes
+        self._sign_modified_obj("snapshot", snapshot_obj)
+        timestamp_meta = self.trusted_set.timestamp.signed.meta["snapshot.json"]
+        true_hashes = timestamp_meta.hashes or {}
+        modified_snapshot_data = json.dumps(snapshot_obj.to_dict()).encode()
+        modified_hashes = self._calculate_modified_hashes(
+            true_hashes, modified_snapshot_data
+        )
+        self.trusted_set.timestamp.signed.meta["snapshot.json"].hashes = modified_hashes
+
+        with self.assertRaises(exceptions.RepositoryError):
+            self.trusted_set.update_snapshot(modified_snapshot_data)
+
+    def test_update_snapshot_after_succesfull_update_new_snapshot_meta_version_different(self):
+        self._setup_update_snapshot_after_successful_update_test()
+        # snapshot.meta["project1"].version != new_snapshot.meta["project1"].version
+        for meta_file_path in self.trusted_set.snapshot.signed.meta.keys():
+            self.trusted_set.snapshot.signed.meta[meta_file_path].version = 2
```

Comment:
```suggestion
for metafile in self.trusted_set.snapshot.signed.meta.values():
    metafile.version += 1
```
or something like it?